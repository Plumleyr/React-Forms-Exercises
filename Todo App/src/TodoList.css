.TodoList-main{
  display: flex;
  justify-content: center;
}

.TodoList-List{
  margin: auto;
  text-align: center;
}