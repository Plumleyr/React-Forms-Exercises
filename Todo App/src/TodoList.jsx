import {useState} from 'react'
import NewTodoForm from './NewTodoForm'
import {v4 as uuid} from "uuid"
import Todo from './Todo'
import './TodoList.css'

const TodoList = () => {
  const [todos, setTodos] = useState([])

  const addTodo = (formData) =>{
    let newTodo = {...formData, id: uuid()}
    setTodos((todos) => [...todos, newTodo])
  };

  const removeTodo = (id) => {
    setTodos(todos.filter((todo) => todo.id !== id));
  }

  return(
    <div className='TodoList-main'>
      <div>
        <NewTodoForm addTodo={addTodo} />
        <div className='TodoList-List'>
          <h1>Todo List</h1>
          <ul>
            {todos.map((todo, idx) => {
              return(
                <li  className='TodoList-li' key={idx}>
                  <Todo 
                    id={todo.id}
                    removeTodo={removeTodo}
                    todo={todo.todo}
                    data-testid={`todo-${idx}`}
                  />
                </li>
              )
            })}
          </ul>
        </div>
      </div>
    </div>
  )
}

export default TodoList