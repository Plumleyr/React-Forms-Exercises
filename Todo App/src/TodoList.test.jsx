import {render, fireEvent, screen } from '@testing-library/react'
import TodoList from './TodoList'
import { expect, it } from 'vitest'

it("renders without crashing", () => {
  render(<TodoList />)
});

it("Matches the snapshot", () => {
  const {asFragment} = render(<TodoList/>)
  expect(asFragment()).toMatchSnapshot();
});

it("creates a todo", () => {
  render(<TodoList/>)
  const todoInput = screen.getByPlaceholderText("todo")
  const createButton = screen.getByText('Create Todo')
  
  fireEvent.change(todoInput, {target: {value: 'Go to gym'}})
  fireEvent.click(createButton)
  
  const todo = screen.getByTestId('todo-0')
  expect(todo).toHaveTextContent('Go to gym')
  
  const deleteButton = screen.getByText('x')
  fireEvent.click(deleteButton)
  expect(todo).not.toBeInTheDocument();
});