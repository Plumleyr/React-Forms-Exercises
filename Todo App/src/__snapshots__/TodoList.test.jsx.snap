// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList-main"
  >
    <div>
      <form>
        <label
          for="todo"
        >
          Todo: 
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="todo"
          type="text"
          value=""
        />
        <button>
          Create Todo
        </button>
      </form>
      <div
        class="TodoList-List"
      >
        <h1>
          Todo List
        </h1>
        <ul />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
